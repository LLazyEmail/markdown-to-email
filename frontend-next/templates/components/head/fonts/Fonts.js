import React from 'react';

const Fonts = () => (                                                                                                                                    // data-testid   | dataTestid                                                        
  <link href="https://fonts.googleapis.com/css?family=Merriweather:400,400i,700,700i|Merriweather+Sans:400,400i,700,700i|Source+Sans+Pro:400,400i,700,700i" data-testid="fontsHrefTest" rel="stylesheet" />
);

export default Fonts;
