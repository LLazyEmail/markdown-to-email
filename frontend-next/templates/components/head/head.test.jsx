import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Head from './head';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Fonts from './fonts';

configure({ adapter: new Adapter() });

describe('head component', () => {

    it('Head component has an <a> element with no empty value and correct attributes', () => {
        const wrapper = shallow(<Head />)
        const trendline = wrapper.find('title');
        expect(trendline.text()).toEqual('The Secrets of High-Performing DevOps teams');
    })

    it('Head component contains <Fonts> component and it renders', () => {
        const wrapper = shallow(<Head />)
        expect(wrapper.contains(<Fonts />)).toEqual(true);
    })

  test('renders without crashing', () => {
    const div = document.createElement('html'); // div was
    ReactDOM.render(<Head />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('renders correctly', () => {
    const tree = create(<Head />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
