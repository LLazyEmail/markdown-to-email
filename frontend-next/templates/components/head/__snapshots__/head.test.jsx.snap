// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`head component renders correctly 1`] = `
<head>
  <meta
    charSet="UTF-8"
  />
  <meta
    content="IE=edge"
    httpEquiv="X-UA-Compatible"
  />
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <title>
    The Secrets of High-Performing DevOps teams
  </title>
  <style />
  <link
    data-testid="fontsHrefTest"
    href="https://fonts.googleapis.com/css?family=Merriweather:400,400i,700,700i|Merriweather+Sans:400,400i,700,700i|Source+Sans+Pro:400,400i,700,700i"
    rel="stylesheet"
  />
</head>
`;
